src/Types/GenericModelPropertyType.php

```diff
@@ -0,0 +1,215 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace NunoMaduro\Larastan\Types;
+
+use NunoMaduro\Larastan\Properties\ModelPropertyExtension;
+use PHPStan\Broker\Broker;
+use PHPStan\TrinaryLogic;
+use PHPStan\Type\ClassStringType;
+use PHPStan\Type\CompoundType;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\Generic\TemplateType;
+use PHPStan\Type\Generic\TemplateTypeMap;
+use PHPStan\Type\Generic\TemplateTypeVariance;
+use PHPStan\Type\IntersectionType;
+use PHPStan\Type\MixedType;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\ObjectWithoutClassType;
+use PHPStan\Type\StaticType;
+use PHPStan\Type\StringType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeCombinator;
+use PHPStan\Type\UnionType;
+use PHPStan\Type\VerbosityLevel;
+
+class GenericModelPropertyType extends ModelPropertyType
+{
+    /** @var Type */
+    private $type;
+
+    /** @var ModelPropertyExtension */
+    protected $propertyExtension;
+
+    public function __construct(Type $type, ModelPropertyExtension $propertyExtension)
```

Comment:
The `ModelPropertyExtension` should really not be in the constructor. But I'm not sure how I can fetch an instance of the extension from a static context?